system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj

```diff
@@ -0,0 +1,108 @@
+        gran (data-core/ingest-concept-with-metadata-file
+                 "echo10-samples/granule-bulk-granule-mime-type-update.echo10"
+                 {:provider-id "PROV1"
+                  :concept-type :granule
+                  :native-id "online-resource-url-update-test-gran2"
+                  :format "application/echo10+xml"})]
+    (swap! test-atom merge {:collection coll
+                            :granule gran})
+    (f)
+    (reset! test-atom {})))
+
+(use-fixtures :each (join-fixtures [(ingest/reset-fixture {"provguid1" "PROV1"})
+                                    ingest-granule-fixture]))
+
+(def no-hits? (complement string/includes?))
+
+(deftest online-resource-url-update-test--echo10--error-cases
+  (testing "invalid update json provided fails"
+    (let [bulk-update-options {:token (echo-util/login (system/context) "user1")}
+          bulk-update {:operation "UPDATE_FIELD"
+                       :update-field "OnlineResourceURL"
+                       :updoots [{"GranuleUR" "mime-type-update-gran-ur"
+                                  "Links" [{:to "http://missing-from"}]}]}
+          {:keys [status task-id]} (ingest/bulk-update-granules
+                                    "PROV1" bulk-update bulk-update-options)
+          {:keys [concept-id revision-id]} (:granule @test-atom)]
+      (index/wait-until-indexed)
+      (ingest/update-granule-bulk-update-task-statuses)
+
+      (is (= 400 status))))
+
+  (testing "no matching url given"
+    (let [bulk-update-options {:token (echo-util/login (system/context) "user1")}
+          bulk-update {:operation "UPDATE_FIELD"
+                       :update-field "OnlineResourceURL"
+                       :updates [{"GranuleUR" "mime-type-update-gran-ur"
+                                  "Links" [{:from "https://no-matching-url"
+                                            :to "ftp://link-1-updated"}]}]}
+          {:keys [status task-id]} (ingest/bulk-update-granules
+                                    "PROV1" bulk-update bulk-update-options)
+          {:keys [concept-id revision-id]} (:granule @test-atom)]
+      (index/wait-until-indexed)
+      (ingest/update-granule-bulk-update-task-statuses)
+
+      ;; verify the granule status is UPDATED
+      (is (= 200 status))
+      (is (some? task-id))
+      (let [status-req-options {:query-params {:show_granules "true"}}
+            status-response (ingest/granule-bulk-update-task-status task-id status-req-options)
+            {:keys [task-status status-message granule-statuses]} status-response]
+        (is (= "COMPLETE" task-status))
+        (is (= "Task completed with 1 FAILED out of 1 total granule update(s)." status-message))
+        (is (= [{:granule-ur "mime-type-update-gran-ur"
+                 :status "FAILED"
+                 :status-message "Update failed - please only specify URLs contained in the existing granule OnlineResources or OnlineAccessURLs [https://no-matching-url] were not found"}]
```

Comment:
From this error message, it seems the update OnlineResourceURL can update either OnlineResources or OnlineAccessURLs. We only have test verifying update OnlineResources URL below. We need to add a test that verifies update OnlineAccessURLs.

Reply:
Fixed wording of the error message